maintenancewindow.go

```diff
@@ -0,0 +1,143 @@
+type MaintenanceWindowOptions struct {
+	// Required. The reason for the maintenance window.
+	Reason string `json:"reason,omitempty"`
+
+	// Required. The title of the maintenance window.
+	Title string `json:"title,omitempty"`
+
+	// Required. The start time of the maintenance window in seconds since 1 Jan 1970
+	StartTimeInSeconds int64 `json:"startTimeInSeconds,omitempty"`
+
+	// Required. The end time of the maintenance window in seconds since 1 Jan 1970
+	EndTimeInSeconds int64 `json:"endTimeInSeconds,omitempty"`
+
+	RelevantCustomerTags []string `json:"relevantCustomerTags"`
+	RelevantHostTags     []string `json:"relevantHostTags,omitempty"`
+	RelevantHostNames    []string `json:"relevantHostNames,omitempty"`
+}
+
+// MaintenanceWindows is used to perform maintenance window related operations
+// against the Wavefront API
+type MaintenanceWindows struct {
+	// client is the Wavefront client used to perform target-related operations
+	client Wavefronter
+}
+
+// MaintenanceWindows is used to return a client for maintenance window
+// related operations
+func (c *Client) MaintenanceWindows() *MaintenanceWindows {
+	return &MaintenanceWindows{client: c}
+}
+
+// GetByID returns the MaintenanceWindow with given ID. If no such
+// MaintenanceWindow exists, GetByID returns an error. The caller can call
+// NotFound on err to determine whether or not the error is because the
+// MaintenanceWindow doesn't exist.
+func (m *MaintenanceWindows) GetByID(id string) (
+	maintenanceWindow *MaintenanceWindow, err error) {
+	var result MaintenanceWindow
+	err = doRest(
+		"GET",
+		fmt.Sprintf("%s/%s", maintenanceWindowEndpoint, id),
+		m.client,
+		doOutput(&result))
+	if err != nil {
+		return
+	}
+	return &result, nil
+}
+
+// Create creates a MaintenanceWindow according to options and returns the
+// newly created MaintenanceWindow.
+func (m *MaintenanceWindows) Create(options *MaintenanceWindowOptions) (
+	maintenanceWindow *MaintenanceWindow, err error) {
+	var result MaintenanceWindow
+	err = doRest(
+		"POST",
+		maintenanceWindowEndpoint,
+		m.client,
+		doInput(options),
+		doOutput(&result))
+	if err != nil {
+		return
+	}
+	return &result, nil
+}
+
+// Update updates a MaintenanceWindow according to options and returns the
+// updated MaintenanceWindow.
+func (m *MaintenanceWindows) Update(
+	id string, options *MaintenanceWindowOptions) (
+	maintenanceWindow *MaintenanceWindow, err error) {
+	var result MaintenanceWindow
+	err = doRest(
+		"PUT",
+		fmt.Sprintf("%s/%s", maintenanceWindowEndpoint, id),
+		m.client,
+		doInput(fixForMonit20607(*options)),
+		doOutput(&result))
+	if err != nil {
+		return
+	}
+	return &result, nil
+}
+
+// DeleteByID deletes the MaintenanceWindow with given id.
+func (m *MaintenanceWindows) DeleteByID(id string) error {
+	return doRest(
+		"DELETE",
+		fmt.Sprintf("%s/%s", maintenanceWindowEndpoint, id),
+		m.client)
+}
+
+func fixForMonit20607(options MaintenanceWindowOptions) *MaintenanceWindowOptions {
```

Comment:
instead of fixForMonit20607, cant we just do a validation/conform function?

Reply:
Not sure what you are suggesting.  Could you provide an example?

As background, RelevantHostTags and RelevantHostNames can be missing from the body of the request, but for some reason if RelevantCustomerTags is missing from the body, it causes a 500 error.  Once 20607 is fixed, this function can be removed.

Reply:
Done. Function renamed.